sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventProcessorClientTest.java

```diff
@@ -648,6 +649,154 @@ public void testSingleEventReceiveHeartBeat() throws InterruptedException {
+     */
+    @Test
+    public void testProcessorDisposeSlow() throws InterruptedException {
+        // Arrange
+        final Tracer tracer1 = mock(Tracer.class);
+        final List<Tracer> tracers = Collections.singletonList(tracer1);
+        TracerProvider tracerProvider = new TracerProvider(tracers);
+
+        when(eventHubClientBuilder.getPrefetchCount()).thenReturn(DEFAULT_PREFETCH_COUNT);
+        when(eventHubClientBuilder.buildAsyncClient()).thenReturn(eventHubAsyncClient);
+        when(eventHubAsyncClient.getFullyQualifiedNamespace()).thenReturn("test-ns");
+        when(eventHubAsyncClient.getEventHubName()).thenReturn("test-eh");
+        when(eventHubAsyncClient.getPartitionIds()).thenReturn(Flux.just("1"));
+        when(eventHubAsyncClient.getIdentifier()).thenReturn("my-client-identifier");
+        when(eventHubAsyncClient
+            .createConsumer(anyString(), anyInt()))
+            .thenReturn(consumer1);
+        when(consumer1.receiveFromPartition(anyString(), any(EventPosition.class), any(ReceiveOptions.class)))
+            .thenReturn(Flux.just(getEvent(eventData1), getEvent(eventData2), getEvent(eventData3)));
+
+        when(eventData1.getSequenceNumber()).thenReturn(1L);
+        when(eventData2.getSequenceNumber()).thenReturn(2L);
+        when(eventData3.getSequenceNumber()).thenReturn(3L);
+        when(eventData1.getOffset()).thenReturn(1L);
+        when(eventData2.getOffset()).thenReturn(100L);
+        when(eventData3.getOffset()).thenReturn(150L);
+
+        final SampleCheckpointStore checkpointStore = new SampleCheckpointStore();
+        // Use 5000 milliseconds to process per event/eventBatch.
+        final MonitorPartitionProcessor testPartitionProcessor = new MonitorPartitionProcessor(5000);
+        CountDownLatch countDownLatch = new CountDownLatch(3);
+        testPartitionProcessor.countDownLatch = countDownLatch;
+
+        final EventProcessorClient eventProcessorClient = new EventProcessorClient(eventHubClientBuilder, "test-consumer",
+            () -> testPartitionProcessor, checkpointStore, false, tracerProvider, ec -> { }, new HashMap<>(),
+            3, Duration.ofSeconds(1), true, Duration.ofSeconds(10), Duration.ofMinutes(1), LoadBalancingStrategy.BALANCED);
+
+        // Act
+        eventProcessorClient.start();
+        boolean completed = countDownLatch.await(10, TimeUnit.SECONDS);
+        eventProcessorClient.stop();
+
+        // Assert
+        assertTrue(completed);
+        // Thread interrupted.
+        assertTrue(testPartitionProcessor.isInterrupted);
+    }
+
+    private static final class MonitorPartitionProcessor extends PartitionProcessor {
+        CountDownLatch countDownLatch;
+        boolean isInterrupted = false;
+        long sleepTime;
+
+        MonitorPartitionProcessor(long sleepTimeInMills) {
+            super();
+            sleepTime = sleepTimeInMills;
```

Comment:
Do the below improve the readability?

1. We could update the Ctr of `MonitorPartitionProcessor` to take an additional parameter `int eventsCount`, which we can use to initialize the countDownLatch (i.e., making the latch a private final)
2. We can add a new method to MonitorPartitionProcessor named `awaitForEvents(int timeoutInSec)`, which we could call between start() and stop().